ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -3263,7 +3269,9 @@ static bool ggml_backend_cuda_device_supports_op(ggml_backend_dev_t dev, const g
 }
 
 static bool ggml_backend_cuda_device_supports_buft(ggml_backend_dev_t dev, ggml_backend_buffer_type_t buft) {
-    return (ggml_backend_buft_is_cuda(buft) || ggml_backend_buft_is_cuda_split(buft)) && buft->device == dev;
+    ggml_backend_cuda_device_context * dev_ctx = (ggml_backend_cuda_device_context *) dev->context;
+    const bool integrated = ggml_cuda_info().devices[dev_ctx->device].integrated;
+    return (ggml_backend_buft_is_cuda(buft) || ggml_backend_buft_is_cuda_split(buft) || (integrated && ggml_backend_buft_is_cuda_host(buft))) && buft->device == dev;
```

Comment:
```suggestion
    return ((ggml_backend_buft_is_cuda(buft) || ggml_backend_buft_is_cuda_split(buft) && buft->device == dev) || (integrated && ggml_backend_buft_is_cuda_host(buft)));
```

I noticed that the logic was not quite correct. The memory needs to either be in VRAM and on the same device (condition on master) or it needs to be in pinned memory for a pinned GPU.